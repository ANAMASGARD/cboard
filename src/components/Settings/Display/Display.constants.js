export const DISPLAY_SIZE_STANDARD = 'Standard';
export const DISPLAY_SIZE_LARGE = 'Large';
export const DISPLAY_SIZE_EXTRALARGE = 'ExtraLarge';
