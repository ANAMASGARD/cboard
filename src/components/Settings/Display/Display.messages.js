import { defineMessages } from 'react-intl';
import {
  DISPLAY_SIZE_STANDARD,
  DISPLAY_SIZE_LARGE,
  DISPLAY_SIZE_EXTRALARGE
} from './Display.constants';

export default defineMessages({
  display: {
    id: 'cboard.components.Settings.Display.display',
    defaultMessage: 'Display'
  },
  [DISPLAY_SIZE_STANDARD]: {
    id: 'cboard.components.Settings.Display.StandardSize',
    defaultMessage: 'Standard'
  },
  [DISPLAY_SIZE_LARGE]: {
    id: 'cboard.components.Settings.Display.LargeSize',
    defaultMessage: 'Large'
  },
  [DISPLAY_SIZE_EXTRALARGE]: {
    id: 'cboard.components.Settings.Display.ExtraLargeSize',
    defaultMessage: 'Extra Large'
  },
  boardElementsSize: {
    id: 'cboard.components.Settings.Display.boardElementsSize',
    defaultMessage: 'Board Elements Size'
  },
  boardElementsSizeSecondary: {
    id: 'cboard.components.Settings.Display.boardElementsSizeSecondary',
    defaultMessage: '...'
  },
  boardFontSize: {
    id: 'cboard.components.Settings.Display.boardFontSize',
    defaultMessage: 'Board Font Size'
  },
  boardFontSizeSecondary: {
    id: 'cboard.components.Settings.Display.boardFontSizeSecondary',
    defaultMessage: '...'
  },
  settingsSize: {
    id: 'cboard.components.Settings.Display.settingsSize',
    defaultMessage: 'Settings Size'
  },
  settingsSizeSecondary: {
    id: 'cboard.components.Settings.Display.settingsSizeSecondary',
    defaultMessage: '...'
  },
  pictogramSize: {
    id: 'cboard.components.Settings.Display.pictogramSize',
    defaultMessage: 'Pictograms Size'
  },
  pictogramSizeSecondary: {
    id: 'cboard.components.Settings.Display.pictogramSizeSecondary',
    defaultMessage: '...'
  }
});
