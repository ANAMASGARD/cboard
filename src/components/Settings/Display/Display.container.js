import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { injectIntl, intlShape } from 'react-intl';
import { updateDisplaySettings } from '../../App/App.actions';
import Display from './Display.component';

export class DisplayContainer extends PureComponent {
  render() {
    const { history } = this.props;
    return <Display {...this.props} onClose={history.goBack} />;
  }
}

DisplayContainer.props = {
  intl: intlShape.isRequired,
  displaySettings: PropTypes.object.isRequired,
  history: PropTypes.object.isRequired,
  updateDisplaySettings: PropTypes.func
};

const mapStateToProps = ({ app: { displaySettings } }) => ({ displaySettings });

const mapDispatchToProps = {
  updateDisplaySettings
};

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(injectIntl(DisplayContainer));
