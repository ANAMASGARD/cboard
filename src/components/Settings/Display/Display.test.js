import React from 'react';
import { shallowMatchSnapshot } from '../../../common/test_utils';

import Display from './Display.component';

jest.mock('./Display.messages', () => {
  return {
    display: {
      id: 'cboard.components.Settings.Display.display',
      defaultMessage: 'Display'
    },
    Standard: {
      id: 'cboard.components.Settings.Display.StandardSize',
      defaultMessage: 'Standard'
    },
    Large: {
      id: 'cboard.components.Settings.Display.LargeSize',
      defaultMessage: 'Large'
    },
    Extralarge: {
      id: 'cboard.components.Settings.Display.ExtraLargeSize',
      defaultMessage: 'Extra Large'
    },
    boardElementsSize: {
      id: 'cboard.components.Settings.Display.boardElementsSize',
      defaultMessage: 'Board Elements Size'
    },
    boardElementsSizeSecondary: {
      id: 'cboard.components.Settings.Display.boardElementsSizeSecondary',
      defaultMessage: '...'
    },
    boardFontSize: {
      id: 'cboard.components.Settings.Display.boardFontSize',
      defaultMessage: 'Board Font Size'
    },
    boardFontSizeSecondary: {
      id: 'cboard.components.Settings.Display.boardFontSizeSecondary',
      defaultMessage: '...'
    },
    settingsSize: {
      id: 'cboard.components.Settings.Display.settingsSize',
      defaultMessage: 'Settings Size'
    },
    settingsSizeSecondary: {
      id: 'cboard.components.Settings.Display.settingsSizeSecondary',
      defaultMessage: '...'
    },
    pictogramSize: {
      id: 'cboard.components.Settings.Display.pictogramSize',
      defaultMessage: 'Pictograms Size'
    },
    pictogramSizeSecondary: {
      id: 'cboard.components.Settings.Display.pictogramSizeSecondary',
      defaultMessage: '...'
    }
  };
});

const COMPONENT_PROPS = {
  displaySettings: {},
  intl: {
    formatMessage: () => {}
  },
  onClose: () => {}
};

describe('Display tests', () => {
  test('default renderer', () => {
    shallowMatchSnapshot(<Display {...COMPONENT_PROPS} />);
  });
});
