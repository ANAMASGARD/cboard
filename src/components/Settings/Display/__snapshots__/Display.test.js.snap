// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display tests default renderer 1`] = `
<ForwardRef(WithStyles)
  onClose={[Function]}
  onSubmit={[Function]}
  open={true}
  title={
    <FormattedMessage
      defaultMessage="Display"
      id="cboard.components.Settings.Display.display"
      values={Object {}}
    />
  }
>
  <ForwardRef(WithStyles)
    className="Display"
  >
    <ForwardRef(WithStyles)>
      <ForwardRef(WithStyles)>
        <ForwardRef(WithStyles)
          className="Display__ListItemText"
          primary={
            <FormattedMessage
              defaultMessage="UI Size"
              id="cboard.components.Settings.Display.uiSize"
              values={Object {}}
            />
          }
          secondary={
            <FormattedMessage
              defaultMessage="Elements size"
              id="cboard.components.Settings.Display.uiSizeSecondary"
              values={Object {}}
            />
          }
        />
        <ForwardRef(WithStyles)
          className="Display__Options"
        >
          <ForwardRef(WithStyles)>
            <ForwardRef(WithStyles)
              aria-label={
                Object {
                  "defaultMessage": "UI Size",
                  "id": "cboard.components.Settings.Display.uiSize",
                }
              }
              id="uiSize"
              name="uiSize"
              onChange={[Function]}
              value="Standard"
            >
              <ForwardRef(WithStyles)
                value="Standard"
              >
                Object {
                  "defaultMessage": "Standard",
                  "id": "cboard.components.Settings.Display.StandardSize",
                }
              </ForwardRef(WithStyles)>
              <ForwardRef(WithStyles)
                value="Large"
              >
                Object {
                  "defaultMessage": "Large",
                  "id": "cboard.components.Settings.Display.LargeSize",
                }
              </ForwardRef(WithStyles)>
              <ForwardRef(WithStyles)
                value="ExtraLarge"
              />
            </ForwardRef(WithStyles)>
          </ForwardRef(WithStyles)>
        </ForwardRef(WithStyles)>
      </ForwardRef(WithStyles)>
      <ForwardRef(WithStyles) />
      <ForwardRef(WithStyles)>
        <ForwardRef(WithStyles)
          className="Display__ListItemText"
          primary={
            <FormattedMessage
              defaultMessage="Font Size"
              id="cboard.components.Settings.Display.fontSize"
              values={Object {}}
            />
          }
          secondary={
            <FormattedMessage
              defaultMessage="App font size"
              id="cboard.components.Settings.Display.fontSizeSecondary"
              values={Object {}}
            />
          }
        />
        <ForwardRef(WithStyles)
          className="Display__Options"
        >
          <ForwardRef(WithStyles)>
            <ForwardRef(WithStyles)
              aria-label={
                Object {
                  "defaultMessage": "Font Size",
                  "id": "cboard.components.Settings.Display.fontSize",
                }
              }
              id="fontSize"
              name="fontSize"
              onChange={[Function]}
              value="Standard"
            >
              <ForwardRef(WithStyles)
                value="Standard"
              >
                Object {
                  "defaultMessage": "Standard",
                  "id": "cboard.components.Settings.Display.StandardSize",
                }
              </ForwardRef(WithStyles)>
              <ForwardRef(WithStyles)
                value="Large"
              >
                Object {
                  "defaultMessage": "Large",
                  "id": "cboard.components.Settings.Display.LargeSize",
                }
              </ForwardRef(WithStyles)>
              <ForwardRef(WithStyles)
                value="ExtraLarge"
              />
            </ForwardRef(WithStyles)>
          </ForwardRef(WithStyles)>
        </ForwardRef(WithStyles)>
      </ForwardRef(WithStyles)>
      <ForwardRef(WithStyles) />
      <ForwardRef(WithStyles)>
        <ForwardRef(WithStyles)
          className="Display__ListItemText"
          primary={
            <FormattedMessage
              values={Object {}}
            />
          }
          secondary={
            <FormattedMessage
              values={Object {}}
            />
          }
        />
        <ForwardRef(WithStyles)>
          <ForwardRef(WithStyles)
            color="secondary"
            onChange={[Function]}
            value="active"
          />
        </ForwardRef(WithStyles)>
      </ForwardRef(WithStyles)>
      <ForwardRef(WithStyles) />
      <ForwardRef(WithStyles)>
        <ForwardRef(WithStyles)
          className="Display__ListItemText"
          primary={
            <FormattedMessage
              values={Object {}}
            />
          }
          secondary={
            <FormattedMessage
              values={Object {}}
            />
          }
        />
        <ForwardRef(WithStyles)
          className="Display__Options"
        >
          <ForwardRef(WithStyles)>
            <ForwardRef(WithStyles)
              id="labelPosition"
              name="labelPosition"
              onChange={[Function]}
            >
              <ForwardRef(WithStyles)
                value="Above"
              />
              <ForwardRef(WithStyles)
                value="Below"
              />
              <ForwardRef(WithStyles)
                value="Hidden"
              />
            </ForwardRef(WithStyles)>
          </ForwardRef(WithStyles)>
        </ForwardRef(WithStyles)>
      </ForwardRef(WithStyles)>
      <ForwardRef(WithStyles) />
      <ForwardRef(WithStyles)>
        <ForwardRef(WithStyles)
          className="Display__ListItemText"
          primary={
            <FormattedMessage
              values={Object {}}
            />
          }
          secondary={
            <FormattedMessage
              values={Object {}}
            />
          }
        />
        <ForwardRef(WithStyles)
          className="Display__Options"
        >
          <ForwardRef(WithStyles)
            color="secondary"
            onChange={[Function]}
            value="active"
          />
        </ForwardRef(WithStyles)>
      </ForwardRef(WithStyles)>
    </ForwardRef(WithStyles)>
  </ForwardRef(WithStyles)>
</ForwardRef(WithStyles)>
`;
